rust/processor/migrations/2024-05-22-200847_add_v1_migration_views/up.sql

```diff
@@ -0,0 +1,319 @@
+-- Your SQL goes here
+
+
+-- Create the schema
+CREATE SCHEMA IF NOT EXISTS legacy_migration_v1;
+
+-- Replace `move_resources` with account transactions
+CREATE OR REPLACE VIEW legacy_migration_v1.move_resources AS
+    SELECT *
+    FROM account_transactions at2
+  ;
+-- account_transactions already has index on transaction_version
+
+-- replace `coin_activities` with `fungible_asset_activities`
+CREATE OR REPLACE VIEW legacy_migration_v1.coin_activities AS
+    SElECT
+        transaction_version,
+        owner_address as event_account_address,
+        -- these two below are mildly concerning
+        0 as event_creation_number,
+        0 as event_sequence_number,
+        owner_address,
+        asset_type AS coin_type,
+        amount,
+        "type" AS activity_type,
+        is_gas_fee,
+        is_transaction_success,
+        entry_function_id_str,
+        block_height,
+        transaction_timestamp,
+        inserted_at,
+        event_index,
+        gas_fee_payer_address,
+        storage_refund_amount,
+    FROM public.fungible_asset_activities
+    WHERE token_standard = 'v1'
+  ;
+CREATE INDEX lm1_ca_ct_a_index ON public.fungible_asset_activities USING btree (asset_type, amount)
+CREATE INDEX lm1_ca_ct_at_a_index ON public.fungible_asset_activities USING btree (asset_type, "type", amount)
+CREATE INDEX lm1_ca_oa_ct_at_index ON public.fungible_asset_activities USING btree (owner_address, asset_type, "type", amount)
+CREATE INDEX lm1_ca_oa_igf_index ON public.fungible_asset_activities USING btree (owner_address, is_gas_fee)
+
+-- replace `coin_balances` with `fungible_asset_balances`
+CREATE OR REPLACE VIEW legacy_migration_v1.coin_balances AS
+    SELECT
+        transaction_version,
+        owner_address,
+        -- this is mainly for hashing the coin type for primary key
+        encode(sha256(asset_type::bytea), 'hex') as coin_type_hash,
```

Comment:
do we actually have sha256 support enabled in alloy?

Reply:
Is the intention to use this as a join? Because if so need to add this as an index on the underlying table

Reply:
I ran this in dbeaver and it worked so it seems like it's supported. Also I forgot to add a migration view for coin_infos, which is the only table this could join with. Yes we should add indexes on the underlying tables.